crates/ruff_linter/src/rules/flake8_bugbear/rules/assert_raises_exception.rs

```diff
@@ -56,6 +56,39 @@ impl fmt::Display for ExceptionKind {
     }
 }
 
+fn detect_blind_exception(
+    semantic: &ruff_python_semantic::SemanticModel<'_>,
+    func: &Expr,
+    arguments: &Arguments,
+) -> Option<ExceptionKind> {
+    let is_assert_raises = matches!(
+        func,
+        &Expr::Attribute(ast::ExprAttribute { ref attr, .. }) if attr.as_str() == "assertRaises"
+    );
+
+    let is_pytest_raises = semantic
+        .resolve_qualified_name(func)
+        .is_some_and(|qualified_name| matches!(qualified_name.segments(), ["pytest", "raises"]));
+
+    if !(is_assert_raises || is_pytest_raises) {
+        return None;
+    }
+
+    if is_pytest_raises && arguments.find_keyword("match").is_some() {
```

Comment:
I think this is supposed to be `is_none` like before right?

```suggestion
    if is_pytest_raises && arguments.find_keyword("match").is_none() {
```

It's a little hard to tell because the diff shifted around, but I'm pretty sure one of the snapshots is showing a different result.

I think this `is_some` is actually correct and the existing `is_none` is a bug unless I'm misunderstanding something, but I'd rather handle that separately, I guess.

Reply:
Unless I'm missing something, I believe `is_some()` is the right test here. `pytest.raises(..., match="pattern")` is a form that already narrows the check to a specific exception message (equivalent to the `assertRaisesRegex` case for `unittest`), so when that `match=` keyword is present we want to *skip* B017, because it’s no longer a “blind” exception assertion.

Reply:
Hmm, maybe I got confused by the logic here. The snapshot on `main` looks correct. Let's regroup the test cases and make sure none of the stable results change, but I think you may be right.

`is_some` definitely makes sense intuitively, I guess the condition was negated in the original code.